src/robotFactory.js

```diff
@@ -1,10 +1,82 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
+    this.coords = coords;
+    this.coords.x = coords.x || 0;
+    this.coords.y = coords.y || 0;
+  }
 
-class FlyingRobot {}
+  goForward(step = 1) {
+    if (step > 0) {
+      this.coords.y += step;
+    }
+  }
 
-class DeliveryDrone {}
+  goBack(step = 1) {
+    if (step > 0) {
+      this.coords.y -= step;
+    }
+  }
+
+  goRight(step = 1) {
+    if (step > 0) {
+      this.coords.x += step;
+    }
+  }
+
+  goLeft(step = 1) {
+    if (step > 0) {
+      this.coords.x -= step;
+    }
+  }
+
+  getInfo() {
+    return `Robot: ${this.name}, Chip version: ${this.chipVersion},`
+     + ` Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+    this.coords.z = coords.z || 0;
+  };
+
+  goDown(step = 1) {
+    if (step > 0) {
+      this.coords.z -= step;
+    }
+  }
+
+  goUp(step = 1) {
+    if (step > 0) {
+      this.coords.z += step;
+    }
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(name, weight, coords, chipVersion,
+    maxLoadWeight, currentLoad = null) {
```

Comment:
Make sure that your arguments are written in one line or each one starts from a new line.

Rewrite this way 👇 
```suggestion
  constructor(name, weight, coords, chipVersion, maxLoadWeight, currentLoad) {
```

Or this way 👇 
```suggestion
  constructor(
    name, 
    weight, 
    coords,
    chipVersion, 
    maxLoadWeight, 
    currentLoad
  ) {
```